airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/SSHTunnel.java

```diff
@@ -0,0 +1,277 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.base;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.io.StringReader;
+import java.net.URISyntaxException;
+import java.security.KeyPair;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import org.apache.sshd.client.SshClient;
+import org.apache.sshd.client.keyverifier.AcceptAllServerKeyVerifier;
+import org.apache.sshd.client.session.ClientSession;
+import org.apache.sshd.common.util.net.SshdSocketAddress;
+import org.apache.sshd.server.forward.AcceptAllForwardingFilter;
+import org.bouncycastle.asn1.x509.SubjectPublicKeyInfo;
+import org.bouncycastle.openssl.PEMKeyPair;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Encapsulates the connection configuration for an ssh tunnel port forward through a proxy/bastion
+ * host plus the remote host and remote port to forward to a specified local port.
+ */
+public class SSHTunnel {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SSHTunnel.class);
+
+  public static final int TIMEOUT_MILLIS = 15000; // 15 seconds
+  private final String method;
+  private final String host;
+  private final String tunnelSshPort;
+  private final String user;
+  private final String sshkey;
+  private final String password;
+  private final String remoteDatabaseHost;
+  private final String remoteDatabasePort;
+  private final String tunnelDatabasePort;
+
+  private SSHTunnel tunnelConfig = null;
+  private SshClient sshclient = null;
+  private ClientSession tunnelSession = null;
+
+  public SSHTunnel(String method,
+                   String host,
+                   String tunnelSshPort,
+                   String user,
+                   String sshkey,
+                   String password,
+                   String remoteDatabaseHost,
+                   String remoteDatabasePort,
+                   String tunnelDatabasePort) {
+    if (method == null) {
+      this.method = "NO_TUNNEL";
+    } else {
+      this.method = method;
+    }
+    this.host = host;
+    this.tunnelSshPort = tunnelSshPort;
+    this.user = user;
+    this.sshkey = sshkey;
+    this.password = password;
+    this.remoteDatabaseHost = remoteDatabaseHost;
+    this.remoteDatabasePort = remoteDatabasePort;
+    this.tunnelDatabasePort = tunnelDatabasePort;
+  }
+
+  public static SSHTunnel getInstance(JsonNode config) {
```

Comment:
Reading through the code is a bit confusing for the case where the `SSHTunnel` doesn't have a valid configuration.

Maybe it'd be easier to read if this was `Optional<SSHTunnel> getInstance(...)`? Then the calls to it could be done with `tunnel.ifPresent(...)` which might read a bit better to show that this is an optional feature.